### PR TITLE
Add basic `SiStripApproximateCluster` Monitoring

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

Title says it all. Provide a new package in order to be able to monitor both online and offline the `SiStripApproximateCluster` collection and created the necessary modifications to be able to run it in the offline DQM by usage of the `approxSiStripClusters` process modifier and in the SiStrip Online DQM client.

#### PR validation:

Run successfully `runTheMatrix.py -l 140.58 -t 4 -j 8 --ibeos` and obtained plots, both with the `compareClusters` parameter set to `True` or `False`.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Not a backport, no backport is needed

### CODE REVIEW

**Comment by mmusich** [comment]
code-checks

**Comment by mmusich** [comment]
test parameters:
   * workflow = 140.58

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by smuzaffar** [comment]
@mmusich , new package `DQM/SiStripMonitorApproximateCluster` just contains single plugin. Why not just include it in `DQM/SiStripMonitorClient`  itself?

**Comment by mmusich** [comment]
> Why not just include it in `DQM/SiStripMonitorClient` itself?

for starters that package is not the right target (it's meant for harvesting codes). Secondly, I think it's worth keeping it separate, also I am expecting this package to grow in time.

File: DQM/SiStripMonitorApproximateCluster/plugins/BuildFile.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<use name="FWCore/Framework"/>
+<use name="FWCore/PluginManager"/>
+<use name="FWCore/ParameterSet"/>
+<use name="DQMServices/Core"/>
```
Comment (makortel) [inline DQM/SiStripMonitorApproximateCluster/plugins/BuildFile.xml:4]: If kept as separate package, the dependencies to `DataFormats/Common` and `DataFormats/SiStripCluster` are missing.

File: DQM/SiStripMonitorApproximateCluster/plugins/SiStripMonitorApproximateCluster.cc [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+//
+//
+
+#include <string>
+
+// user include files
+#include "DQMServices/Core/interface/DQMEDAnalyzer.h"
+#include "DQMServices/Core/interface/MonitorElement.h"
+#include "DataFormats/Common/interface/DetSet.h"
+#include "DataFormats/Common/interface/DetSetVector.h"
```
Comment (makortel) [inline DQM/SiStripMonitorApproximateCluster/plugins/SiStripMonitorApproximateCluster.cc:25]: This `#include` seems to be unused.

File: DQM/SiStripMonitorApproximateCluster/python/SiStripMonitorApproximateCluster_cfi.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import FWCore.ParameterSet.Config as cms
+
+# SiStripMonitorCluster
+from DQMServices.Core.DQMEDAnalyzer import DQMEDAnalyzer
+SiStripMonitorApproximateCluster = DQMEDAnalyzer("SiStripMonitorApproximateCluster",
```
Comment (makortel) [inline DQM/SiStripMonitorApproximateCluster/python/SiStripMonitorApproximateCluster_cfi.py:5]: Why not rely on the generated `cfi` file?

File: DQM/SiStripMonitorApproximateCluster/plugins/BuildFile.xml (hunk 1 above)
Comment (mmusich) [reply DQM/SiStripMonitorApproximateCluster/plugins/BuildFile.xml]: Addressed at https://github.com/cms-sw/cmssw/pull/40295/commits/a90f8e55543f72001dfd4eafd3a64d2a4b362cd5

File: DQM/SiStripMonitorApproximateCluster/plugins/SiStripMonitorApproximateCluster.cc (hunk 2 above)
Comment (mmusich) [reply DQM/SiStripMonitorApproximateCluster/plugins/SiStripMonitorApproximateCluster.cc]: addressed at https://github.com/cms-sw/cmssw/pull/40295/commits/a90f8e55543f72001dfd4eafd3a64d2a4b362cd5

File: DQM/SiStripMonitorApproximateCluster/python/SiStripMonitorApproximateCluster_cfi.py (hunk 3 above)
Comment (mmusich) [reply DQM/SiStripMonitorApproximateCluster/python/SiStripMonitorApproximateCluster_cfi.py]: Addressed at https://github.com/cms-sw/cmssw/pull/40295/commits/a90f8e55543f72001dfd4eafd3a64d2a4b362cd5

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
@cms-sw/dqm-l2 sorry for multiple pushes. Should be ready for review now.

**Comment by mmusich** [comment]
type trk

**Comment by emanueleusai** [comment]
+1

- spurious DQM differences

**Comment by rappoccio** [comment]
+1

Changes in DQM unrelated to this PR.

**Pull request merged**